holidays/countries/india.py

```diff
@@ -28,269 +37,598 @@ class India(HolidayBase, ChristianHolidays, InternationalHolidays, IslamicHolida
-        self._add_holiday_mar_30("Rajasthan Day")
-        self._add_holiday_jun_15("Maharana Pratap Jayanti")
+        # Dr. B. R. Ambedkar Jayanti.
+        self._add_holiday_apr_14(tr("Dr. B. R. Ambedkar's Jayanti"))
+        # Maharana Pratap Jayanti.
+        self._add_holiday_jun_15(tr("Maharana Pratap Jayanti"))
+        # Rajasthan Day.
+        self._add_holiday_mar_30(tr("Rajasthan Day"))
 
     # Sikkim.
     def _populate_subdiv_sk_public_holidays(self):
-        self._add_holiday_may_16("Annexation Day")
+        # Dr. B. R. Ambedkar Jayanti.
+        self._add_holiday_apr_14(tr("Dr. B. R. Ambedkar's Jayanti"))
+        # Sikkim State Day.
+        self._add_holiday_may_16(tr("Sikkim State Day"))
 
     # Tamil Nadu.
     def _populate_subdiv_tn_public_holidays(self):
-        self._add_holiday_apr_14("Dr. B. R. Ambedkar's Jayanti")
-        self._add_holiday_apr_14("Puthandu (Tamil New Year)")
-        self._add_holiday_apr_15("Puthandu (Tamil New Year)")
+        # Dr. B. R. Ambedkar Jayanti.
+        self._add_holiday_apr_14(tr("Dr. B. R. Ambedkar's Jayanti"))
+        # Pongal.
+        self._add_makar_sankranti(tr("Pongal"))
+        # Puthandu.
+        self._add_holiday_apr_14(tr("Puthandu (Tamil New Year)"))
 
     # Telangana.
     def _populate_subdiv_ts_public_holidays(self):
-        self._add_holiday_apr_6("Eid al-Fitr")
-        self._add_holiday_oct_6("Bathukamma Festival")
+        # Bathukamma Festival.
+        self._add_holiday_oct_6(tr("Bathukamma Festival"))
+        # Dr. B. R. Ambedkar Jayanti.
+        self._add_holiday_apr_14(tr("Dr. B. R. Ambedkar's Jayanti"))
+        # Telangana Formation Day.
+        self._add_holiday_jun_2(tr("Telangana Formation Day"))
 
     # Uttarakhand.
     def _populate_subdiv_uk_public_holidays(self):
-        self._add_holiday_apr_14("Dr. B. R. Ambedkar's Jayanti")
+        # Dr. B. R. Ambedkar Jayanti.
+        self._add_holiday_apr_14(tr("Dr. B. R. Ambedkar's Jayanti"))
 
     # Uttar Pradesh.
     def _populate_subdiv_up_public_holidays(self):
-        self._add_holiday_apr_14("Dr. B. R. Ambedkar's Jayanti")
+        # Chhath Puja.
+        self._add_chhath_puja(tr("Chhath Puja"))
+        # Dr. B. R. Ambedkar Jayanti.
+        self._add_holiday_apr_14(tr("Dr. B. R. Ambedkar's Jayanti"))
+        # UP Formation Day.
+        self._add_holiday_jan_24(tr("UP Formation Day"))
 
     # West Bengal.
     def _populate_subdiv_wb_public_holidays(self):
-        self._add_holiday_apr_14("Dr. B. R. Ambedkar's Jayanti")
-        self._add_holiday_apr_14("Pohela Boishakh")
-        self._add_holiday_apr_15("Pohela Boishakh")
-        self._add_holiday_may_9("Rabindra Jayanti")
+        # Dr. B. R. Ambedkar Jayanti.
+        self._add_holiday_apr_14(tr("Dr. B. R. Ambedkar's Jayanti"))
+        # Pohela Boisakh.
+        self._add_holiday_apr_14(tr("Pohela Boishakh"))
+        # Pohela Boisakh.
+        self._add_holiday_apr_15(tr("Pohela Boishakh"))
```

Comment:
```suggestion
        # Pohela Boisakh.
        self._add_holiday_apr_15(tr("Pohela Boishakh"))
```
The bot recommends leaving only one day so strongly that I checked the sources. And indeed, according to [West Bengal government](https://finance.wb.gov.in/writereaddata/4712-F%28P2%29.pdf), Bengali New Year (Pohela_Boishakh) is observed as single day, Apr 15.